src/moveCues.ts

```diff
@@ -0,0 +1,93 @@
+import { Rundown, RundownCueOrderItem } from '@rundown-studio/types'
+
+/**
+ * Moves selected cues to a new position within the rundown's cue order.
+ *
+ * @async
+ * @function
+ * @param {RundownCueOrderItem[]} cueOrder - The original array of cue order items.
+ * @param {Array<Rundown['id']>} selectedCues - IDs of the cues to move.
+ * @param {string} destination - Target position formatted as "mainIndex.subIndex".
+ * @throws {Error} If a group header is attempted to be moved inside another group.
+ * @return {Promise<RundownCueOrderItem[]>} - The reordered cue array.
+ */
+export async function moveCues (
+  cueOrder: RundownCueOrderItem[],
+  selectedCues: Array<Rundown['id']>,
+  destination: string,
```

Comment:
Conceptual question: If this is always a 'move after' operation, why not use the cueId as desination?

Reply:
You can send 0 for the beginning or a bigger number than the last cue to send to start or end.

Also, as the index is not saved anywhere, I will have to loop through it or in the client, in the functions or here, so I'd rather only do it here and keep it tested.

Reply:
Ah yes, makes sense